src/SystemTextJson.JsonDiffPatch/Diffs/JsonDiffPatcher.Array.cs

```diff
@@ -121,7 +121,7 @@ static partial class JsonDiffPatcher
 
                         // We have two objects equal by position or other criteria
                         var itemDiff = new JsonDiffDelta();
-                        DiffInternal(ref itemDiff, left[entry.LeftIndex], right[entry.RightIndex], options);
+                        DiffInternal(ref itemDiff, left[commonHead + entry.LeftIndex], right[commonHead + entry.RightIndex], options);
```

Comment:
Thanks for digging. 

I think this change makes sense. I realized I couldn't reproduce this issue without a custom key finder. Is that what you use? Without a key finder the diff result is correct but not optimal (no modification).

Do you want to add a test for this?

Reply:
Indeed using an array objectkeyfinder, added a specific unittest that covers this usecase.